### DISCUSSION TITLE
Blitz Storage & File Uploads

### DISCUSSION THREAD

**Discussion by estevanjantsk** [blitz-js/blitz#1785] [Ideas]
I love the idea that blitz brings to us to facilitate things. One ideia that I would love to see is somthing like this:
https://edgeguides.rubyonrails.org/active_storage_overview.html

Active Storage facilitates uploading files to a cloud storage service like Amazon S3, Google Cloud Storage, or Microsoft Azure Storage and attaching those files to Active Record objects. It comes with a local disk-based service for development and testing and supports mirroring files to subordinate services for backups and migrations.

I really don't know how to do this in Blitz but it's only an ideia. Would be almost like the same as we have with the authentication/authorization that we get out of the box. Create some mutations to do some file(s) upload? In development we would save in the public folder, for ex.

Something like strapi does with their api. When you create a model and this model has one atribute with type file or multifile it already exposes some mutation method that you can use to upload the file. https://strapi.io/documentation/developer-docs/latest/plugins/upload.html#upload-files-related-to-an-entry

**Comment by flybayer** [comment]
Thanks @estevanjantsk, this is a good request. Likely I won't be able to give attention to this until after 1.0, but anyone else is welcome to dig in and come up with a proposal/prototype!

**Reply by Zeko369** [reply]
I'd like to take this and setup something that would work with local storage (for server deployments), any S3 comaptible API (AWS, Google, Microsoft, DO...), and maybe have some specific opt in stuff for images through Cloudinary?

**Comment by garytube** [comment]
I find the  [Strapi Upload Providers](https://strapi.io/documentation/developer-docs/latest/plugins/upload.html#using-a-provider) pretty nice. Maybe this can be used as reference point

**Comment by flybayer** [comment]
For reference, here is a early draft of an abandoned image upload RFC: https://gist.github.com/flybayer/8590c423a6c4a94f8efacfd4d6a61064
